d3a_interface/sim_results/kpi.py

```diff
@@ -145,11 +146,59 @@ def update_area_kpi(self, area_dict, core_stats):
         self._accumulate_energy_trace(core_stats)
 
 
+class SavingKPI:
+    def __init__(self):
+        self.producer_list = list()
+        self.consumer_list = list()
+        self.ess_list = list()
+        self.house_list = list()
+        self.fit_revenue = 0.
+        self.utility_bill = 0.
+        self.d3a_revenue = 0.
+        self.saving_absolute = 0.
+        self.saving_percentage = 0.
+
+    def calculate_saving_kpi(self, area_dict, core_stats, gf_alp):
+        """
+        :param area_dict: contain nested area info
+        :param core_stats: contain area's raw/key statistics
+        :param gf_alp: grid_fee_along_the_path - cumulative grid fee from root to target area
+        :return:
+        """
+        for child in area_dict['children']:
+            if is_producer_node_type(child):
+                if_not_in_list_append(self.producer_list, child['uuid'])
+            elif is_load_node_type(child):
+                if_not_in_list_append(self.consumer_list, child['uuid'])
+            elif is_prosumer_node_type(child):
+                if_not_in_list_append(self.ess_list, child['uuid'])
+
+        # fir_excl_gf_alp: feed-in tariff rate excluding grid fee along path
+        fir_excl_gf_alp = (core_stats.get(area_dict['uuid'], {}).get('feed_in_tariff', 0.) -
+                           gf_alp)
+        # mmr_incl_gf_alp: market maker rate include grid fee along path
+        mmr_incl_gf_alp = (core_stats.get(area_dict['uuid'], {}).get('market_maker_rate', 0.) +
+                           gf_alp)
```

Comment:
I think that, instead of using acronyms, a much easier approach for the reader would be to use separate small functions, that are self-documented by the function name. That way you will not need the acronyms and just name the variables as: 
`feed_in_tariff = self.get_feed_in_tariff_rate_excluding_path_grid_fees(core_stats, area_dict["uuid"], path_grid_fee)`